Extension/src/LanguageServer/configurations.ts

```diff
@@ -998,10 +998,15 @@ export class CppProperties {
                 } else if (configuration.includePath) {
                     // If the user doesn't set browse.path, copy the includePath over. Make sure ${workspaceFolder} is in there though...
                     configuration.browse.path = configuration.includePath.slice(0);
-                    if (configuration.includePath.findIndex((value: string) =>
-                        !!value.match(/^\$\{(workspaceRoot|workspaceFolder)\}(\\\*{0,2}|\/\*{0,2})?$/g)) === -1
-                    ) {
-                        configuration.browse.path.push("${workspaceFolder}");
+                    if (this.rootUri) {
```

Comment:
Do we know if this behavior changed due to a change in vscode itself, or something else?

if it was a change in vscode, did we want to still make it work with `(workspaceRoot|workspaceFolder)` as well in case the newer extension was run on an older version of vscode?

Reply:
It needs to continue to work with `${workspaceFolder}/**` because that's the default entry for includePath.

Reply:
@fearthecowboy @bobbrow The change is that this the `${workspaceFolder}` variable is resolved at this point by our own code due to calls to the resolveVariable or whatever it's called.

Reply:
`resolveVariables` does not resolve `${workspaceFolder}`. That happens in the native code.

Reply:
@bobbrow No, that changed. It resolves it in TypeScript. That's what I saw when I debugged it.

Reply:
Where? I reviewed the implementation of `resolveVariables` before I made that comment and the behavior has not changed.

https://github.com/microsoft/vscode-cpptools/blob/db1cfda601b6bdc7463edf37d9fbcecb880920c2/Extension/src/common.ts#L429-L431

Reply:
There is also `resolvePath` which will replace the ${workspaceFolder} variable, but its return value is only supposed to be used as a temporary, e.g. for configuration validation and setting up watchers for compile commands.

Reply:
@bobbrow Yes, it's resolvePath, called by resolveAndSplit.

Reply:
Called by updateConfigurationPathsArray, which is called by updateServerOnFolderSettingsChange, which is the function that was modified in this PR.

Reply:
@bobbrow I recall Colen talking about changing it to not send ${workspaceFolder} to the cpptools side and resolving it on the TypeScript side instead, since it wasn't necessary keep ${workspaceFolder} around.

Reply:
Well, I'm not sure when that changed, but I suppose if it has, then yes this code will need to be updated to deal with it. But I think it doesn't make a lot of sense to be inserting `${workspaceFolder}` at a point when `${workspaceFolder}` is expected to have been resolved away, right?  We're in a halfway state now and that causes confusion.

Reply:
@bobbrow Sure, but my PR just fixes it so that it doesn't add ${workspaceFolder} -- the pre-existing behavior already adds workspace folder. And note I've moved this to draft since Colen says one of his pending PRs would be causing this code to not be used so it might not be needed.

Reply:
The native side will resolve variables (`${workspaceRoot}`, `${workspaceFolder}`, `${vcpkgRoot}`) when ingesting browse paths (and include paths, mac framework paths, forced includes, compile commands paths, compiler path, compiler args, and the database filename) from the base configuration.  Perhaps processing was added on the TypeScript side to address needing one of these values (compiler path in a build task maybe?) with it already resolved.